### ISSUE TITLE
Large kmem_alloc fails at spl_kmem_zalloc

### ISSUE THREAD

**Issue by marcelhuberfoo** [zfsonlinux/zfs#3429]
Mai 18 21:40:47 vaio kernel: Large kmem_alloc(42672, 0x1000), please file an issue at:
                             https://github.com/zfsonlinux/zfs/issues/new
Mai 18 21:40:50 vaio kernel: CPU: 0 PID: 194 Comm: zvol/16 Tainted: P           O    4.0.3-1-ARCH #1
Mai 18 21:40:50 vaio kernel: Hardware name: Sony Corporation SVP1321C5E/VAIO, BIOS R1040V7 09/09/2013
Mai 18 21:40:50 vaio kernel:  0000000000000000 0000000061622a2f ffff8802137d7ba8 ffffffff81571bd3
Mai 18 21:40:50 vaio kernel:  0000000000000000 000000000000c210 ffff8802137d7bf8 ffffffffa02dfc94
Mai 18 21:40:50 vaio kernel:  ffff8801c30d3af8 0000000000000017 ffff8802137d7be8 ffff8801c30d3af8
Mai 18 21:40:50 vaio kernel: Call Trace:
Mai 18 21:40:50 vaio kernel:  [<ffffffff81571bd3>] dump_stack+0x4c/0x6e
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02dfc94>] spl_kmem_zalloc+0x164/0x1e0 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04523fb>] dmu_buf_hold_array_by_dnode+0x9b/0x5c0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa0452a13>] dmu_buf_hold_array+0x73/0xb0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04542ca>] dmu_write_req+0x8a/0x1f0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa050029b>] zvol_write+0x10b/0x460 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3501>] taskq_thread+0x1c1/0x380 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff810a15a0>] ? wake_up_process+0x50/0x50
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3340>] ? taskq_cancel_id+0x140/0x140 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093418>] kthread+0xd8/0xf0
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel:  [<ffffffff81577418>] ret_from_fork+0x58/0x90
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel: Possible memory allocation deadlock: size=42672 lflags=0xc210
Mai 18 21:40:50 vaio kernel: CPU: 0 PID: 194 Comm: zvol/16 Tainted: P           O    4.0.3-1-ARCH #1
Mai 18 21:40:50 vaio kernel: Hardware name: Sony Corporation SVP1321C5E/VAIO, BIOS R1040V7 09/09/2013
Mai 18 21:40:50 vaio kernel:  0000000000000000 0000000061622a2f ffff8802137d7ba8 ffffffff81571bd3
Mai 18 21:40:50 vaio kernel:  0000000000000000 000000000000c210 ffff8802137d7bf8 ffffffffa02dfcff
Mai 18 21:40:50 vaio kernel:  ffff8801c30d3af8 0000000000000017 ffff8802137d7be8 ffff8801c30d3af8
Mai 18 21:40:50 vaio kernel: Call Trace:
Mai 18 21:40:50 vaio kernel:  [<ffffffff81571bd3>] dump_stack+0x4c/0x6e
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02dfcff>] spl_kmem_zalloc+0x1cf/0x1e0 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04523fb>] dmu_buf_hold_array_by_dnode+0x9b/0x5c0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa0452a13>] dmu_buf_hold_array+0x73/0xb0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04542ca>] dmu_write_req+0x8a/0x1f0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa050029b>] zvol_write+0x10b/0x460 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3501>] taskq_thread+0x1c1/0x380 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff810a15a0>] ? wake_up_process+0x50/0x50
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3340>] ? taskq_cancel_id+0x140/0x140 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093418>] kthread+0xd8/0xf0
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel:  [<ffffffff81577418>] ret_from_fork+0x58/0x90
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel: Possible memory allocation deadlock: size=42672 lflags=0xc210
Mai 18 21:40:50 vaio kernel: CPU: 0 PID: 194 Comm: zvol/16 Tainted: P           O    4.0.3-1-ARCH #1
Mai 18 21:40:50 vaio kernel: Hardware name: Sony Corporation SVP1321C5E/VAIO, BIOS R1040V7 09/09/2013
Mai 18 21:40:50 vaio kernel:  0000000000000000 0000000061622a2f ffff8802137d7ba8 ffffffff81571bd3
Mai 18 21:40:50 vaio kernel:  0000000000000000 000000000000c210 ffff8802137d7bf8 ffffffffa02dfcff
Mai 18 21:40:50 vaio kernel:  ffff8801c30d3af8 0000000000000017 ffff8802137d7be8 ffff8801c30d3af8
Mai 18 21:40:50 vaio kernel: Call Trace:
Mai 18 21:40:50 vaio kernel:  [<ffffffff81571bd3>] dump_stack+0x4c/0x6e
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02dfcff>] spl_kmem_zalloc+0x1cf/0x1e0 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04523fb>] dmu_buf_hold_array_by_dnode+0x9b/0x5c0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa0452a13>] dmu_buf_hold_array+0x73/0xb0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04542ca>] dmu_write_req+0x8a/0x1f0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa050029b>] zvol_write+0x10b/0x460 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3501>] taskq_thread+0x1c1/0x380 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff810a15a0>] ? wake_up_process+0x50/0x50
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3340>] ? taskq_cancel_id+0x140/0x140 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093418>] kthread+0xd8/0xf0
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel:  [<ffffffff81577418>] ret_from_fork+0x58/0x90
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel: Possible memory allocation deadlock: size=42672 lflags=0xc210
Mai 18 21:40:50 vaio kernel: CPU: 0 PID: 194 Comm: zvol/16 Tainted: P           O    4.0.3-1-ARCH #1
Mai 18 21:40:50 vaio kernel: Hardware name: Sony Corporation SVP1321C5E/VAIO, BIOS R1040V7 09/09/2013
Mai 18 21:40:50 vaio kernel:  0000000000000000 0000000061622a2f ffff8802137d7ba8 ffffffff81571bd3
Mai 18 21:40:50 vaio kernel:  0000000000000000 000000000000c210 ffff8802137d7bf8 ffffffffa02dfcff
Mai 18 21:40:50 vaio kernel:  ffff8801c30d3af8 0000000000000017 ffff8802137d7be8 ffff8801c30d3af8
Mai 18 21:40:50 vaio kernel: Call Trace:
Mai 18 21:40:50 vaio kernel:  [<ffffffff81571bd3>] dump_stack+0x4c/0x6e
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02dfcff>] spl_kmem_zalloc+0x1cf/0x1e0 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04523fb>] dmu_buf_hold_array_by_dnode+0x9b/0x5c0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa0452a13>] dmu_buf_hold_array+0x73/0xb0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04542ca>] dmu_write_req+0x8a/0x1f0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa050029b>] zvol_write+0x10b/0x460 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3501>] taskq_thread+0x1c1/0x380 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff810a15a0>] ? wake_up_process+0x50/0x50
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3340>] ? taskq_cancel_id+0x140/0x140 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093418>] kthread+0xd8/0xf0
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel:  [<ffffffff81577418>] ret_from_fork+0x58/0x90
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel: Possible memory allocation deadlock: size=42672 lflags=0xc210
Mai 18 21:40:50 vaio kernel: CPU: 0 PID: 194 Comm: zvol/16 Tainted: P           O    4.0.3-1-ARCH #1
Mai 18 21:40:50 vaio kernel: Hardware name: Sony Corporation SVP1321C5E/VAIO, BIOS R1040V7 09/09/2013
Mai 18 21:40:50 vaio kernel:  0000000000000000 0000000061622a2f ffff8802137d7ba8 ffffffff81571bd3
Mai 18 21:40:50 vaio kernel:  0000000000000000 000000000000c210 ffff8802137d7bf8 ffffffffa02dfcff
Mai 18 21:40:50 vaio kernel:  ffff8801c30d3af8 0000000000000017 ffff8802137d7be8 ffff8801c30d3af8
Mai 18 21:40:50 vaio kernel: Call Trace:
Mai 18 21:40:50 vaio kernel:  [<ffffffff81571bd3>] dump_stack+0x4c/0x6e
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02dfcff>] spl_kmem_zalloc+0x1cf/0x1e0 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04523fb>] dmu_buf_hold_array_by_dnode+0x9b/0x5c0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa0452a13>] dmu_buf_hold_array+0x73/0xb0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa04542ca>] dmu_write_req+0x8a/0x1f0 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa050029b>] zvol_write+0x10b/0x460 [zfs]
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3501>] taskq_thread+0x1c1/0x380 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff810a15a0>] ? wake_up_process+0x50/0x50
Mai 18 21:40:50 vaio kernel:  [<ffffffffa02e3340>] ? taskq_cancel_id+0x140/0x140 [spl]
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093418>] kthread+0xd8/0xf0
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 18 21:40:50 vaio kernel:  [<ffffffff81577418>] ret_from_fork+0x58/0x90
Mai 18 21:40:50 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170

**Comment by behlendorf** [comment]
@marcelhuberfoo thanks for reporting this.  By chance have you enabled large block support?  If so, that's fine, I'm just trying to explain the large allocation size.

**Comment by marcelhuberfoo** [comment]
@behlendorf Unfortunately I checked **after** the kernel panic and detected a non-upgraded pool:
```
~ $ sudo zpool status -v
  pool: foopool
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
	still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
	the pool may no longer be accessible by software that does not support
	the features. See zpool-features(5) for details.
  scan: scrub repaired 0 in 0h10m with 0 errors on Mon May 18 08:00:25 2015
config:

	NAME         STATE     READ WRITE CKSUM
	foopool      ONLINE       0     0     0
	  cryptroot  ONLINE       0     0     0

errors: No known data errors

~ $ sudo zpool upgrade -a
This system supports ZFS pool feature flags.

Enabled the following features on 'foopool':
  filesystem_limits
  large_blocks
```
So the crash might have happened due to missing features.

**Comment by marcelhuberfoo** [comment]
Sorry, I forgot to mention the versions of spl and zfs I am using:
spl: 0.6.4.4.g62e2eb2; means 4 commits ahead of tag 0.6.4, commitid: 62e2eb2
zfs: 0.6.4.71.g7fec46b; same structure as above, 71 commits ahead of 0.6.4 tag

**Comment by marcelhuberfoo** [comment]
Here is another dump with probably similar origin but this one was catched with all features enabled on the pool:
```
Mai 19 10:00:13 vaio kernel: spl_kmem_alloc_impl: 29806 callbacks suppressed
Mai 19 10:00:13 vaio kernel: Large kmem_alloc(54224, 0x1000), please file an issue at:
                             https://github.com/zfsonlinux/zfs/issues/new
Mai 19 10:00:13 vaio kernel: CPU: 3 PID: 199 Comm: zvol/21 Tainted: P           O    4.0.3-1-ARCH #1
Mai 19 10:00:13 vaio kernel: Hardware name: Sony Corporation SVP1321C5E/VAIO, BIOS R1040V7 09/09/2013
Mai 19 10:00:13 vaio kernel:  0000000000000000 00000000f0799557 ffff8802137fbba8 ffffffff81571bd3
Mai 19 10:00:13 vaio kernel:  0000000000000000 000000000000c210 ffff8802137fbbf8 ffffffffa02dfc94
Mai 19 10:00:13 vaio kernel:  ffff8801c30d3af8 0000000000000017 ffff8802137fbbe8 ffff8801c30d3af8
Mai 19 10:00:13 vaio kernel: Call Trace:
Mai 19 10:00:13 vaio kernel:  [<ffffffff81571bd3>] dump_stack+0x4c/0x6e
Mai 19 10:00:13 vaio kernel:  [<ffffffffa02dfc94>] spl_kmem_zalloc+0x164/0x1e0 [spl]
Mai 19 10:00:13 vaio kernel:  [<ffffffffa04523fb>] dmu_buf_hold_array_by_dnode+0x9b/0x5c0 [zfs]
Mai 19 10:00:13 vaio kernel:  [<ffffffffa0452a13>] dmu_buf_hold_array+0x73/0xb0 [zfs]
Mai 19 10:00:13 vaio kernel:  [<ffffffffa04542ca>] dmu_write_req+0x8a/0x1f0 [zfs]
Mai 19 10:00:13 vaio kernel:  [<ffffffffa050029b>] zvol_write+0x10b/0x460 [zfs]
Mai 19 10:00:13 vaio kernel:  [<ffffffffa02e3501>] taskq_thread+0x1c1/0x380 [spl]
Mai 19 10:00:13 vaio kernel:  [<ffffffff810a15a0>] ? wake_up_process+0x50/0x50
Mai 19 10:00:13 vaio kernel:  [<ffffffffa02e3340>] ? taskq_cancel_id+0x140/0x140 [spl]
Mai 19 10:00:13 vaio kernel:  [<ffffffff81093418>] kthread+0xd8/0xf0
Mai 19 10:00:13 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
Mai 19 10:00:13 vaio kernel:  [<ffffffff81577418>] ret_from_fork+0x58/0x90
Mai 19 10:00:13 vaio kernel:  [<ffffffff81093340>] ? kthread_worker_fn+0x170/0x170
```

**Comment by drescherjm** [comment]
I am seeing this too. 

[10642.880532] Large kmem_alloc(65536, 0x1000), please file an issue at:
https://github.com/zfsonlinux/zfs/issues/new
[10642.880537] CPU: 0 PID: 2766 Comm: zvol/12 Tainted: P        W  O    4.0.4-gentoo-jmd0.comcast.net-20150518 #1
[10642.880539] Hardware name: System manufacturer System Product Name/P5Q-PRO, BIOS 1306    08/20/2008
[10642.880540]  0000000000010000 ffff8800c30bbbf8 ffffffff815d6909 000000000000001a
[10642.880544]  0000000000000000 ffff8800c30bbc38 ffffffffa041db7b ffff8800c30bbcd0
[10642.880547]  ffff880100ed1848 00000050e8000000 0000000000000000 0000000004000000
[10642.880550] Call Trace:
[10642.880553]  [<ffffffff815d6909>] dump_stack+0x45/0x57
[10642.880561]  [<ffffffffa041db7b>] spl_kmem_zalloc+0x11b/0x190 [spl]
[10642.880575]  [<ffffffffa04c4e5f>] dmu_buf_rele_array+0xaf/0x520 [zfs]
[10642.880590]  [<ffffffffa04c53a8>] dmu_buf_hold_array_by_bonus+0xd8/0x100 [zfs]
[10642.880605]  [<ffffffffa04c6a02>] dmu_write_req+0x62/0x1d0 [zfs]
[10642.880627]  [<ffffffffa0566c01>] zrl_is_locked+0x1111/0x18e0 [zfs]
[10642.880631]  [<ffffffff815d81ab>] ? __schedule+0x32b/0x890
[10642.880638]  [<ffffffffa0421088>] taskq_cancel_id+0x2c8/0x470 [spl]
[10642.880641]  [<ffffffff81067f40>] ? wake_up_process+0x40/0x40
[10642.880648]  [<ffffffffa0420ee0>] ? taskq_cancel_id+0x120/0x470 [spl]
[10642.880651]  [<ffffffff8105ebb4>] kthread+0xc4/0xe0
[10642.880654]  [<ffffffff8105eaf0>] ? kthread_create_on_node+0x170/0x170
[10642.880657]  [<ffffffff815dba08>] ret_from_fork+0x58/0x90
[10642.880660]  [<ffffffff8105eaf0>] ? kthread_create_on_node+0x170/0x170
[11208.997631] EXT4-fs (zd0): mounted filesystem with ordered data mode. Opts: (null)
[11423.283988] Large kmem_alloc(64992, 0x1000), please file an issue at:
https://github.com/zfsonlinux/zfs/issues/new
[11423.283996] CPU: 2 PID: 2760 Comm: zvol/6 Tainted: P        W  O    4.0.4-gentoo-jmd0.comcast.net-20150518 #1
[11423.283998] Hardware name: System manufacturer System Product Name/P5Q-PRO, BIOS 1306    08/20/2008
[11423.284016]  000000000000fde0 ffff8800c30afbf8 ffffffff815d6909 0000000000000001
[11423.284020]  0000000000000000 ffff8800c30afc38 ffffffffa041db7b ffff8800c30afcd0
[11423.284023]  ffff880048f81f38 0000003ddca89000 0000000000000000 0000000003f77000
[11423.284027] Call Trace:
[11423.284035]  [<ffffffff815d6909>] dump_stack+0x45/0x57
[11423.284062]  [<ffffffffa041db7b>] spl_kmem_zalloc+0x11b/0x190 [spl]
[11423.284088]  [<ffffffffa04c4e5f>] dmu_buf_rele_array+0xaf/0x520 [zfs]
[11423.284104]  [<ffffffffa04c53a8>] dmu_buf_hold_array_by_bonus+0xd8/0x100 [zfs]
[11423.284120]  [<ffffffffa04c6a02>] dmu_write_req+0x62/0x1d0 [zfs]
[11423.284142]  [<ffffffffa0566c01>] zrl_is_locked+0x1111/0x18e0 [zfs]
[11423.284161]  [<ffffffff815d81ab>] ? __schedule+0x32b/0x890
[11423.284175]  [<ffffffffa0421088>] taskq_cancel_id+0x2c8/0x470 [spl]
[11423.284181]  [<ffffffff81067f40>] ? wake_up_process+0x40/0x40
[11423.284187]  [<ffffffffa0420ee0>] ? taskq_cancel_id+0x120/0x470 [spl]
[11423.284192]  [<ffffffff8105ebb4>] kthread+0xc4/0xe0
[11423.284195]  [<ffffffff8105eaf0>] ? kthread_create_on_node+0x170/0x170
[11423.284199]  [<ffffffff815dba08>] ret_from_fork+0x58/0x90
[11423.284202]  [<ffffffff8105eaf0>] ? kthread_create_on_node+0x170/0x170

and I have not upgraded my pools.

jmd0 ~ # zpool status
  pool: zfs_2t0
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
        still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(5) for details.
  scan: scrub repaired 0 in 4h26m with 0 errors on Thu Jun  5 14:23:53 2014
config:

        NAME        STATE     READ WRITE CKSUM
        zfs_2t0     ONLINE       0     0     0
          sdg2      ONLINE       0     0     0

errors: No known data errors

  pool: zfs_4t0
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
        still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(5) for details.
  scan: scrub repaired 0 in 5h50m with 0 errors on Thu Jun  5 15:48:06 2014
config:

        NAME        STATE     READ WRITE CKSUM
        zfs_4t0     ONLINE       0     0     0
          sdb       ONLINE       0     0     0

errors: No known data errors

  pool: zfs_home
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
        still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(5) for details.
  scan: none requested
config:

        NAME        STATE     READ WRITE CKSUM
        zfs_home    ONLINE       0     0     0
          sda3      ONLINE       0     0     0

errors: No known data errors

jmd0 ~ # cat /sys/module/{spl,zfs}/version
0.6.4-4_g62e2eb2
0.6.4-74_g141b638

**Comment by diametric** [comment]
I noticed a similar issue trying to debug a problem where I am unable to mount zvols at all.  Does not matter the filesystem (ext3, 4, ntfs, etc).   The actual mount() system call succeeds, but the filesystem is not mounted.  I can however mount non-ZFS volumes such as USB disks, or regular ZFS datasets. I'm not sure if this is related or not.

[85038.324068] Large kmem_alloc(65536, 0x1000), please file an issue at:
               https://github.com/zfsonlinux/zfs/issues/new
[85038.324075] CPU: 0 PID: 199 Comm: zvol/19 Tainted: P           O    4.0.1-1-ARCH #1
[85038.324077] Hardware name: ASUSTeK COMPUTER INC. P9D-C Series/P9D-C Series, BIOS 0904 03/27/2014
[85038.324079]  0000000000000000 00000000b87a7be8 ffff880803b3bba8 ffffffff8156de97
[85038.324082]  0000000000000000 000000000000c210 ffff880803b3bbf8 ffffffffa001ac94
[85038.324085]  ffff8807a985f278 0000000000000017 ffff880803b3bbe8 ffff8807a985f278
[85038.324088] Call Trace:
[85038.324096]  [<ffffffff8156de97>] dump_stack+0x4c/0x6e
[85038.324128]  [<ffffffffa001ac94>] spl_kmem_zalloc+0x164/0x1e0 [spl]
[85038.324146]  [<ffffffffa02423fb>] dmu_buf_hold_array_by_dnode+0x9b/0x5c0 [zfs]
[85038.324169]  [<ffffffffa0242a13>] dmu_buf_hold_array+0x73/0xb0 [zfs]
[85038.324182]  [<ffffffffa02442ca>] dmu_write_req+0x8a/0x1f0 [zfs]
[85038.324197]  [<ffffffffa02f003b>] zvol_write+0x10b/0x460 [zfs]
[85038.324204]  [<ffffffffa001e501>] taskq_thread+0x1c1/0x380 [spl]
[85038.324210]  [<ffffffff810a1440>] ? wake_up_process+0x50/0x50
[85038.324216]  [<ffffffffa001e340>] ? taskq_cancel_id+0x140/0x140 [spl]
[85038.324220]  [<ffffffff81093338>] kthread+0xd8/0xf0
[85038.324223]  [<ffffffff81093260>] ? kthread_worker_fn+0x170/0x170
[85038.324227]  [<ffffffff81573718>] ret_from_fork+0x58/0x90
[85038.324229]  [<ffffffff81093260>] ? kthread_worker_fn+0x170/0x170
[85038.326859] Large kmem_alloc(65536, 0x1000), please file an issue at:
               https://github.com/zfsonlinux/zfs/issues/new
[85038.326865] CPU: 4 PID: 191 Comm: zvol/11 Tainted: P           O    4.0.1-1-ARCH #1
[85038.326867] Hardware name: ASUSTeK COMPUTER INC. P9D-C Series/P9D-C Series, BIOS 0904 03/27/2014
[85038.326868]  0000000000000000 0000000080e9c963 ffff880803b0bba8 ffffffff8156de97
[85038.326871]  0000000000000000 000000000000c210 ffff880803b0bbf8 ffffffffa001ac94
[85038.326873]  ffff8807a985f278 0000000000000017 ffff880803b0bbe8 ffff8807a985f278
[85038.326876] Call Trace:
[85038.326883]  [<ffffffff8156de97>] dump_stack+0x4c/0x6e
[85038.326908]  [<ffffffffa001ac94>] spl_kmem_zalloc+0x164/0x1e0 [spl]
[85038.326927]  [<ffffffffa02423fb>] dmu_buf_hold_array_by_dnode+0x9b/0x5c0 [zfs]
[85038.326946]  [<ffffffffa0242a13>] dmu_buf_hold_array+0x73/0xb0 [zfs]
[85038.326960]  [<ffffffffa02442ca>] dmu_write_req+0x8a/0x1f0 [zfs]
[85038.326973]  [<ffffffffa02f003b>] zvol_write+0x10b/0x460 [zfs]
[85038.326981]  [<ffffffffa001e501>] taskq_thread+0x1c1/0x380 [spl]
[85038.326987]  [<ffffffff810a1440>] ? wake_up_process+0x50/0x50
[85038.326993]  [<ffffffffa001e340>] ? taskq_cancel_id+0x140/0x140 [spl]
[85038.326997]  [<ffffffff81093338>] kthread+0xd8/0xf0
[85038.327000]  [<ffffffff81093260>] ? kthread_worker_fn+0x170/0x170
[85038.327004]  [<ffffffff81573718>] ret_from_fork+0x58/0x90
[85038.327007]  [<ffffffff81093260>] ? kthread_worker_fn+0x170/0x170

Running Arch Linux with 4.0.1-1-ARCH kernel, and ZFS from git at commit #63b33e8 and SPL from git commit #62e2eb2

**Comment by prometheanfire** [comment]
should this be combined with #3684

**Comment by behlendorf** [comment]
@prometheanfire yes, yes it should.  Thanks for noticing this one, this is a duplicate of #3684 and was resolved by commit c495fe2c1c6b1c63aefcd832e2e0eb0a20d4c4dc.
